### PR TITLE
Update golangci-lint in Makefile

### PR DESCRIPTION
**Pull request by Gusted**
- Partially resolvess #17596
- Download specific version(v1.43.0) by default.
- If current installed version is older than the minium version, it will download the mininium required version.
- Update the install script to avoid deprecated error `golangci/golangci-lint err this script is deprecated, please do not use it anymore. check https://github.com/goreleaser/godownloader/issues/207`

### CODE REVIEW

**Comment by silverwind** [comment]
I think we should do away with globally installed modules. Does golang offer a solution to install such devDependencies locally yet (Without fake imports)?

**Comment by Gusted** [comment]
> Does golang offer a solution to install such devDependencies locally yet (Without fake imports)?

Does golang offer such solution officially? No, it doesn't.
Does golang support such solution possibly? Yeah-ish, see https://github.com/golang/go/issues/25922#issuecomment-756675225 it's the best thing you can get, but it feels like a hack for me.

**Comment by silverwind** [comment]
Yeah, such fake imports are unacceptable I think because it will download/build these dependencies even on regular builds.

**Comment by Gusted** [comment]
> Yeah, such fake imports are unacceptable I think because it will download/build these dependencies even on regular builds.

Not if you have a custom `XXX.mod` file like mentioned, it will only download the `go.mod`'s dependencies by default.

**Comment by silverwind** [comment]
For the version check, why not do it like it's done for go/node? It's much simpler and should suffice:

https://github.com/go-gitea/gitea/blob/ab1379743e812eb1430e4d15a00c3f7b9e759da6/Makefile#L200-L205

**Comment by Gusted** [comment]
> For the version check, why not do it like it's done for go/node? It's much simpler and should suffice:
> 
> https://github.com/go-gitea/gitea/blob/ab1379743e812eb1430e4d15a00c3f7b9e759da6/Makefile#L200-L205

Not sure how I missed that part, the version check is now simplified.

**Comment by Gusted** [comment]
> > Yeah, such fake imports are unacceptable I think because it will download/build these dependencies even on regular builds.
> 
> Not if you have a custom `XXX.mod` file like mentioned, it will only download the `go.mod`'s dependencies by default.

NVM that, it still requires to have a file with fake imports.

**Comment by silverwind** [comment]
Download does not work, it tries on non-existing URL `https://raw.githubusercontent.com/golangci/golangci-lint/001043000`.

**Comment by Gusted** [comment]
> This will also need to be updated in the following repo: https://gitea.com/gitea/test-env/

https://gitea.com/gitea/test-env/pulls/10

**Comment by lunny** [comment]
> https://gitea.com/gitea/test-env/pulls/10

merged.

File: Makefile [hunk 1]
```diff
@@ -765,12 +766,22 @@ pr\#%: clean-all
...
 .PHONY: golangci-lint
-golangci-lint:
+golangci-lint: golangci-lint-check
+	golangci-lint run --timeout 10m
+
+.PHONY: golangci-lint-check
+golangci-lint-check:
+	$(eval GOLANGCI_LINT_VERSION := $(shell printf "%03d%03d%03d" $(shell golangci-lint --version | grep -Eo '[0-9]+\.[0-9.]+' | tr '.' ' ');))
 	@hash golangci-lint > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
+		echo "Downloading golangci-lint"; \
```
Comment (zeripath) [inline Makefile]: might be useful to add the version:

```suggestion
		echo "Downloading golangci-lint v${MIN_GOLANGCI_LINT_VERSION}"; \
```

**Review by zeripath — APPROVED** [review]
Otherwise LGTM

File: Makefile (hunk 1 above)
Comment (Gusted) [reply Makefile]: That's useful, thanks!

**Comment by silverwind** [comment]
It bothers me a bit that the checks work differently the go/node which are defined with zero-padded numbers. Maybe make those checks match by either formatting/de-formatting them all?

**Comment by Gusted** [comment]
Voila - done.

**Review by silverwind — APPROVED** [review]

**Pull request merged**